File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;

class Solution {
	public String solution(String[] cards1, String[] cards2, String[] goal) {
		String answer = "";

		Queue<String> q1 = new LinkedList<>(Arrays.asList(cards1));
		Queue<String> q2 = new LinkedList<>(Arrays.asList(cards2));
		Queue<String> gq = new LinkedList<>(Arrays.asList(goal));
		
		for(int i = 0; i < goal.length; i++) {
			if(gq.peek().equals(q1.peek())) {
				q1.poll();
				gq.poll();
			}
			else if(gq.peek().equals(q2.peek())) {
				q2.poll();
				gq.poll();
			}
		}
		
		if(gq.isEmpty()) answer = "Yes";
		else answer = "No";
		
		return answer;
	}


}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal : list[str]):
    # 카드 배열을 순회하기 위한 인덱스
    pointer1, pointer2 = 0, 0
    # 바라보고 있는 goal 의 인덱스
    current = 0

    # 반복 종료를 제어하기 위한 플래그
    isMatchCard = False

    while current < len(goal):
        isMatchCard = False

        if pointer1 < len(cards1) and cards1[pointer1] == goal[current]:
            pointer1 += 1
            current += 1
            isMatchCard = True
                            
        if pointer2 < len(cards2) and cards2[pointer2] == goal[current]:
            pointer2 += 1
            current += 1
            isMatchCard = True
        
        if isMatchCard == False:
            break

    return "Yes" if len(goal) == current else "No"